fsspec/caching.py

```diff
@@ -472,7 +533,12 @@ def __init__(self, blocksize, fetcher, size, data={}, strict=True, **_):
         else:
             self.data = data
 
-    def _fetch(self, start, stop):
+    def _fetch(self, start: int | None, stop: int | None) -> bytes:
+        if start is None:
```

Comment:
Sort of. I'm pretty confident that passing in `start=None` would be cause a crash previously (since the `loc <= start < loc1` would error out), so in that sense it's not just to appease mypy.

But making making `start: int | None` (instead of just saying `start: int`) *is* to appease mypy to obey subtyping rules -- (basically: subtypes must have "at least as broad" type signatures as the base class, otherwise if I had `cache: BaseCache; cache._fethc(None, None)` this woudl crash if `cache` was a `KnownPartsOfAFile`).

Mypy's FAQ is decent here https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides.